pkg/zeek/interval.go

```diff
@@ -44,8 +46,8 @@ func (i Interval) String() string {
 	// This format of a fractional second is used by zeek in logs.
 	// It uses enough precision to fully represent the 64-bit ns
 	// accuracy of a nano Duration. Such values cannot be represented by
-	// float64's without loss of the least significant digits of ns,
-	return nano.DurationString(int64(i))
+	// float64's without loss of the least significant digits of ns.
+	return strconv.FormatFloat(time.Duration(i).Seconds(), 'f', 6, 64)
```

Comment:
You can't use IEEE float without loss of information per the comment here.

Reply:
If I'm trying to limit decimal precision to 6 (which I believe zeek logs only support) aren't I encountering information loss anyways?

Reply:
Open to suggestions...

Reply:
Zeek uses IEEE 64-bit floating point for times so limiting it to microseconds is a problem.  Moreover, nano.Ts is even more precise than 64-bit float.  My suggestion would be to change zson.Record.ZeekStrings to take a precision argument and for the zeek printer to use precision 6 until if finds something more granular then switch to precision 9 and it could simply override the String methods for type time.  We can also change the package nano methods to drop the trailing zeros.